### ISSUE TITLE
Use correct booking messages for on demand taxis

### ISSUE THREAD

**Issue by leonardehrenfried** [stadtnavi/digitransit-ui#590]
An example of an on-demand taxi is here: https://dev.stadtnavi.eu/reiseplan/Nufringen%3A%3A48.6224388%2C8.8881021/G%C3%BCltstein%3A%3A48.575747449999994%2C8.878864665540831/onDemandTaxi?time=1623442026

![Screenshot from 2021-06-11 22-07-55](https://user-images.githubusercontent.com/151346/121743328-7c8db680-cb01-11eb-9659-974e757bbf78.png)

The messages to display have slightly moved in the backend therefore you need to adjust the GraphQL query a little and use different fields.

A complete query is this:
```
query{
  plan(
    date: "2021-06-03"
    time: "22:30"
    from: { lat:  48.6226, lon: 8.8879 }
    to: { lat: 48.5755, lon: 8.8795 }
    transportModes: [
      { mode: RAIL },
      { mode: FLEX, qualifier: EGRESS },
      { mode: FLEX, qualifier: DIRECT },
      { mode: WALK },
    ],
    maxWalkDistance: 5000
  ) {
    itineraries {
      legs {
        mode
        rentedBike
        distance
        startTime
        endTime
        
        dropOffBookingInfo {
          message
          dropOffMessage
          contactInfo {
            phoneNumber
          }
        }
        route {
          url
          mode
          shortName
          type
        }
        trip {
          gtfsId
          tripShortName
          
        }
        from {
          name
          bikeRentalStation {
            name
            id
          }
        }
        to {
          name
          bikeRentalStation {
            name
            id
          }
        }
      }
    }
  }
}
```

Please use the fields
- `leg.dropOffBookingInfo.message`
- `leg.dropOffBookingInfo.dropOffMessage`

They should be something like:

"Anmeldung bis 30min vor Abfahrt."
"Haustürbeförderung beim Absetzen. Aufpreis 0,80€."

![Screenshot from 2021-06-08 16-02-07](https://user-images.githubusercontent.com/151346/121199100-f0fef600-c872-11eb-89f2-095a1c1934dc.png)

**Comment by leonardehrenfried** [comment]
Also, `route.type` is correctly set to `715` which means "On demand bus": https://developers.google.com/transit/gtfs/reference/extended-route-types

![Screenshot from 2021-06-09 08-46-47](https://user-images.githubusercontent.com/151346/121306631-60222c00-c8ff-11eb-8e14-b8a598bb92d4.png)

**Comment by leonardehrenfried** [comment]
Right now the check if it's an on demand taxi works by looking at the route name (should start with RT) but it's more reliable to check if the route type is 715.

**Comment by leonardehrenfried** [comment]
can you also do that?

**Comment by Beck-berry** [comment]
Of course!

**Comment by Beck-berry** [comment]
Will the backend still send the `contactInfo` fields `infoUrl` and `bookingUrl`?

**Comment by leonardehrenfried** [comment]
Yes but I think one of them is null. But you don't need to change that part

**Comment by ahmed-nawar** [comment]
@leonardehrenfried 
There are many technical points that I cannot validate. Currently the shown message is like the screen shot.
![image](https://user-images.githubusercontent.com/84456072/122255300-48820f00-cece-11eb-97b2-b32e34eae385.png)

Can you check if all the other technical points are done? And please tell me if there's anything else I should test to close this issue.

**Comment by leonardehrenfried** [comment]
Yes, it's correct.

**Comment by ahmed-nawar** [comment]
The message is removed. Is that correct ? 
![image](https://user-images.githubusercontent.com/84456072/122914458-f40de200-d35a-11eb-86fb-38a190129795.png)

**Comment by leonardehrenfried** [comment]
No, that shouldn't be that way. It's a bug.

**Comment by leonardehrenfried** [comment]
@derhuerst Can you let us know here when a feed with the the pickup and dropoff rules on the regular, scheduled trip are available?

**Comment by leonardehrenfried** [comment]
@ahmed-nawar A bit of background: we have the regular "bus" view which doesn't have the message anymore (or it never had it) but we also have the "to door service" view, which has it.

https://dev.stadtnavi.eu/reiseplan/Ehningen%3A%3A48.6592556%2C8.9405097/G%C3%BCltstein%3A%3A48.575747449999994%2C8.878864665540831/onDemandTaxi/0?time=1624392000

**Comment by derhuerst** [comment]
> Can you let us know here when a feed with the the pickup and dropoff rules on the regular, scheduled trip are available?

I have fixed this in https://github.com/derhuerst/generate-herrenberg-gtfs-flex/commit/e2c1c7f959e30e83a5afe92af0b6c39b5f82509a, currently building a new `hbg5` feed.

**Comment by derhuerst** [comment]
http://116.203.235.64:8002/hbg5.merged.with_flex.gtfs.zip now contains this fix, `shasum` is `85a3bcebb32732b52a06ebb41599655aeb0f4889`.
